include/s3select_functions.h

```diff
@@ -151,18 +170,27 @@ class __function : public base_statement
   }
 
 
-  std::vector<base_statement*> get_arguments()
+  bs_stmt_vec_t& get_arguments()
   {
     return arguments;
   }
 
-  virtual ~__function()
-  {
-    arguments.clear();
-  }
+  virtual ~__function() {}
 };
 
 
+  void s3select_functions::clean()
+  {
+    for(auto d : __all_query_functions)
+    {
+      if (d->is_function())
+      {
+        dynamic_cast<__function*>(d)->impl()->__call_destructor();
```

Comment:
adding a new syntax "__call_destructor" that just calls "->~func()" is superfluous;  just call the dtor--every C++ programmer is familiar with it.

Reply:
ok, will rename.

the s3-select-function framework is calling to the base virtual function(__cal_dest...) , which call base::~dtor() (and that triggered the derived dtor())
meaning, in that way it's transparent to new derived object ( such as s3-select operator / function )